ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -109,6 +115,9 @@
         }
       }
     }
+    for (ReplicaId replicaId : localReplicas) {
```

Comment:
this can be done in one shot: `replicaPool.addAll(0, localReplicas)`.

I was wondering if this is done better by adding things to `replicaPool` directly thus (not considering down replicas):

```java
if (local_replica) {
  replicaPool.addFirst(replica);
} else if (remote_ originating) {
  replicaPool.addLast(replica);
} else {
  remoteNonOriginatingPool.add(replica);
}

if (shouldIncludeNonOriginating) { // a boolean set by a config in your next patch
  replicaPool.addAll(remoteNonOriginatingPool);
}

replicaPool.addAll(downReplicas);
```